quinn/src/lib.rs

```diff
@@ -256,29 +258,20 @@ impl<'a> EndpointBuilder<'a> {
         }
     }
 
-    pub fn reactor(&mut self, handle: &'a tokio_reactor::Handle) -> &mut Self {
-        self.reactor = Some(handle);
-        self
-    }
-    pub fn logger(&mut self, logger: Logger) -> &mut Self {
-        self.logger = logger;
-        self
-    }
-
-    /// Accept incoming connections.
-    pub fn listen(&mut self, config: ServerConfig) -> &mut Self {
-        self.server_config = Some(config);
-        self
+    /// Build an endpoint bound to an unspecified free port.
+    ///
+    /// A good default behavior for clients.
+    pub fn build(self) -> Result<(Endpoint, Driver, Incoming), Error> {
```

Comment:
So I think this means clients using this API won't work on IPv4, right? That seems a little unfriendly.

Reply:
On Linux, IPv6 sockets are dual-stack by default, which means this should Just Work. Unfortunately it appears that [Windows differs here](https://docs.microsoft.com/en-us/windows/desktop/winsock/dual-stack-sockets). OpenBSD does not support dual-stack mode at all, and I'm not sure what macos does.

Forcing IPv4 also doesn't seem ideal; I think the best compromise would be to find a portable way to request dual-stack mode and let OpenBSD users run two endpoints if they really need to. We could probably support associating multiple sockets with a single endpoint if absolutely necessary, though.

Reply:
Can you write up a unit test that validates this? We can then at least know whether it works correctly on Linux, I can test on macOS and at some point we could bring up CI for Windows.

(Maybe this should be spun off into a separate issue, don't want to derail this PR too much.)